### PR TITLE
feat: allow to set a empty state for index table

### PR DESCRIPTION
**Pull request by sifthedog**
- Adds possibility to pass an empty state to index table

### CODE REVIEW

File: app/components/polaris/index_table_component.html.erb [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-            <th class="Polaris-IndexTable__TableHeading">
-              <%= column.title %>
-            </th>
-          <% end %>
-        </tr>
-      </thead>
-      <tbody>
-        <% @data.each do |row| %>
-          <%= render Polaris::BaseComponent.new(**row_arguments(row)) do %>
+  <% if @data.empty? %>
```
Comment (sifthedog) [inline app/components/polaris/index_table_component.html.erb:2]: I am too lazy to do this everytime on my app's side

**Comment by sifthedog** [comment]
I plan on making more additions down the road to building my Shopify app. Let me know if contributions are welcome and what needs addressing, @kirillplatonov

File: app/components/polaris/index_table_component.html.erb (hunk 1 above)
Comment (kirillplatonov) [reply app/components/polaris/index_table_component.html.erb:2]: Perfect motivation to improve it 🙂

**Review by kirillplatonov — APPROVED** [review]
Looks good to me! 👍

Thank you for your contribution. More PRs are welcome!

**Pull request merged**